openadapt/events.py

```diff
@@ -567,31 +610,55 @@ def include_merged_events(to_merge):
 
 
 def discard_unused_events(
-    referred_events, action_events, referred_timestamp_key,
-):
-    referred_event_timestamps = set([
-        getattr(action_event, referred_timestamp_key)
-        for action_event in action_events
-    ])
+    referred_events: list,
+    action_events: list,
```

Comment:
```
referred_events: list[ActionEvent],
action_events: list[ActionEvent],
```

Reply:
Addressed in https://github.com/OpenAdaptAI/OpenAdapt/pull/289/commits/d88767ff785961d7eb36ca589d96f707a019b507